aws/lambda/pytorch-auto-revert/pytorch_auto_revert/testers/hud.py

```diff
@@ -1,94 +1,76 @@
+import json
 import logging
-from typing import Iterable, Optional
+from typing import Any, Mapping, Optional, Union
 
-from ..hud_renderer import build_grid_model, render_html
-from ..signal_extraction import SignalExtractor
+from ..hud_renderer import render_html_from_state
```

Comment:
hud_renderer generates html by concatenation, 

I believe we should move away from this, this is NOT maintainable or easy to understand. Feels vibecoded.

please replace with a proper template compiler engine. It is OK to have the template as a string in a file. but build HTML as string concatenation is something that people made fun of in the 90s :P